### PR TITLE
Extend Load Balancer creation waiter to look for errors in `Errors` field

### PR DESCRIPTION
**Pull request by joaopalet**

### CODE REVIEW

File: services/loadbalancer/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## v0.9.3 (2024-03-19)
+
+- Update Load Balancer creation waiter to check for errors the instance `Errors` field and, if some are found, return even if the status is still `STATUS_PENDING`
```
Comment (vicentepinto98) [inline services/loadbalancer/CHANGELOG.md:3]:
```suggestion
- Improve error handling in Load Balancer creation waiter, fixing timeout for `STATUS_PENDING` state with errors. If an error is found in the `Errors` field, it returns `true` and an error. 
```

File: services/loadbalancer/wait/wait.go [hunk 2]
```diff
@@ -46,6 +47,15 @@ func CreateLoadBalancerWaitHandler(ctx context.Context, a APIClientInterface, pr
 		if s == nil || s.Name == nil || *s.Name != instanceName || s.Status == nil {
 			return false, nil, nil
 		}
+
+		if s.Errors != nil && len(*s.Errors) != 0 {
```
Comment (vicentepinto98) [inline services/loadbalancer/wait/wait.go:51]: Could it happen that the instance has STATUS_READY and errors in the `Errors` field? In that case the waiter will fail.

We should clarify with the team if that can happen and what should we do in that case.

File: services/loadbalancer/wait/wait.go [hunk 3]
```diff
@@ -46,6 +47,15 @@ func CreateLoadBalancerWaitHandler(ctx context.Context, a APIClientInterface, pr
...
+			errors := make([]string, len(*s.Errors))
+			for _, err := range *s.Errors {
+				errors = append(errors, fmt.Sprintf("%s: %s", *err.Type, *err.Description))
+			}
+			return true, s, fmt.Errorf("create failed for instance with name %s, got errors: %s", instanceName, strings.Join(errors, ";"))
```
Comment (vicentepinto98) [inline services/loadbalancer/wait/wait.go:56]: I would also add the status to the error message

File: services/loadbalancer/wait/wait.go (hunk 2 above)
Comment (joaopalet) [reply services/loadbalancer/wait/wait.go]: Discussed it with colleague from the Load Balancer team. They think there is also an issue on their side and will investigate why errors might be set and not be reflected on the state. Nevertheless, he said we could use this errors list to determine if there is an error or not

**Review by vicentepinto98 — APPROVED** [review]

**Pull request merged**